pages/app/app-context.tsx

```diff
@@ -67,6 +69,8 @@ export function AppContextProvider({ children }: { children: React.ReactNode })
   const { mode, pageId } = match ? match.params : { mode: undefined, pageId: undefined };
   const urlParams = parseQuery(location.search) as AppUrlParams;
 
+  document.querySelector('html')?.setAttribute('dir', urlParams.direction);
```

Comment:
This is React. 

In React you do not do such thing bluntly inside the render function. Even if it seems working, it will cause you a hard time dealing with issues later

Reply:
Per offline discussion we are moving forward with a page refresh requirement, and following the implementation for visual refresh: `document.body.classList.toggle('awsui-visual-refresh', visualRefresh);`